go-controller/pkg/ovn/pods.go

```diff
@@ -245,19 +245,41 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
 	portName := fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
 	logrus.Debugf("Creating logical port for %s on switch %s", portName, logicalSwitch)
 
-	out, stderr, err = util.RunOVNNbctl("--wait=sb", "--",
-		"--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-addresses",
-		portName, "dynamic", "--", "set",
-		"logical_switch_port", portName,
-		"external-ids:namespace="+pod.Namespace,
-		"external-ids:logical_switch="+logicalSwitch,
-		"external-ids:pod=true")
-	if err != nil {
-		logrus.Errorf("Error while creating logical port %s "+
-			"stdout: %q, stderr: %q (%v)",
-			portName, out, stderr, err)
-		return
+	annotation, err := util.UnmarshalPodAnnotation(pod.Annotations["ovn"])
+
+	// If pod already has annotations, just add the lsp with static ip/mac.
+	// Else, create the lsp with dynamic addresses.
+	if err == nil {
+		out, stderr, err = util.RunOVNNbctl("--may-exist", "lsp-add",
+			logicalSwitch, portName, "--", "lsp-set-addresses", portName,
+			fmt.Sprintf("%s %s", annotation.MAC, annotation.IP), "--", "set",
```

Comment:
@alexanderConstantinescu I think we need to do `annotation.IP.IP` here. Otherwise it'll write "1.2.3.4/24" which isn't what we expect in GetPortAddresses().

Reply:
Done, thanks for the remark on that.